### PR TITLE
firstboot: Use --prompt in systemd-firstboot.service

### PR DESCRIPTION
**Pull request by daandemeyer**
I was wondering what the reasoning was behind only prompting for locale, timezone and root password. Shouldn't we prompt for everything that isn't initialized?

### CODE REVIEW

**Comment by poettering** [comment]
So the reason I didn't do that initially is that it's a bit weird asking about kbd mappings when we run inside a container where that's worthless...

Maybe we should simply tweak `--prompt` a tiny bit: imply `--prompt-keymap` only if detect_container() finds nothing.

**Comment by poettering** [comment]
Also, I think we should try harder to avoid questions not worth asking. I.e. we should show a menu of shells available (i.e. parse `/etc/shells` if it exists), and if we determine that there are no shells to choose from, then suppress the prompt entirely. That way, if you run a minimal image with just one shell installed the question is not asked.

If /etc/shells doesn't exist I think we should check for /bin/sh and /bin/bash but no others, since they are kinda special on Linux as being the only ones likely there on 90% of systems, and we make references to both of them elsewhere already. Going for a manual hunt for other shells sounds wrong, people should instead populate /etc/shells. (that said, maybe we should also look into /usr/lib/shells as a fallback, because this shouldn't require /etc).

Anyway, long story short: I am fine with turning more of these questions by default on, but we should try really hard to avoid asking them if pointless and that means specifically, not ask them if there's only one choice possible and also not asking them if the environment makes them unnecessary.

Does that make sense?

**Comment by DaanDeMeyer** [comment]
> Also, I think we should try harder to avoid questions not worth asking. I.e. we should show a menu of shells available (i.e. parse `/etc/shells` if it exists), and if we determine that there are no shells to choose from, then suppress the prompt entirely. That way, if you run a minimal image with just one shell installed the question is not asked.

I had actually added some extra logic to the --root-shell stuff that printed a list of available shells but then I realized `getusershell()` and friends don't have a `root` argument so I was just using /etc/shells from the host. It should be easy enough to write a replacement that supports operating on a sysroot and then print a list of options, each with an associated number that the user can choose from. If there's only one, we pick that without asking.

Printing all possible keymaps/locales as options might spam the output a bit as there's so many of them. I personally don't mind but might be something we'd want to avoid.

I'll see if I can find some time to implement this (with the `detect_container` stuff as well), shouldn't be too hard.

**Comment by DaanDeMeyer** [comment]
I'll close this for now and make a new PR once I have free time.
